### DISCUSSION TITLE
Batched field resolver

### DISCUSSION THREAD

**Discussion by PaulFranken** [Netflix/dgs-framework#778] [Q&A]
Hi,

I'm trying to figure out wether it is possible within DGS to include/exclude fields based on a check, without having to execute that check on every entity in the request.

For example, retrieving a list of 10.000 objects and doing a single check to exclude a property of those objects based on the user type, thereby preventing the querying of the child objects from the database on every single object. 

Basically the functionality of the @skip and @include directives.

**Comment by srinivasankavitha** [comment]
The @skip and @include directives are supported by graphql-java already and
the framework should allow you to use them as well.

**Reply by PaulFranken** [reply]
That would still require a check to be done per entity, in this case 10.000 times, right?

**Comment by srinivasankavitha** [comment]
Yes, that would happen each time the resolver/fetcher for the parent of the
field is called, if @skip and @include functionality is what you are
looking for. The other option would be to populate all the fields at the
parent level where you can better control what is fetched instead. You can
use the look ahead functionality by inspecting the data fetching
environment to see what fields are requested, if that is what you are using
to check.
https://www.graphql-java.com/blog/deep-dive-data-fetcher-results/
